sharedbuffers/mapped_struct.py

```diff
@@ -950,12 +950,25 @@ def _stable_hash(key):
         except OverflowError:
             hval = key & 0xFFFFFFFFFFFFFFFF
     elif isinstance(key, float):
-        trunc_key = int(key)
-        if trunc_key == key:
-            hval = trunc_key
-        else:
+        truncated = False
+        try:
+            trunc_key = int(key)
+            if trunc_key == key:
+                hval = trunc_key
+                truncated = True
+        except (OverflowError, ValueError):
+            pass
+
+        if not truncated:
             mant, expo = math.frexp(key)
-            hval = _mix_hash(expo, cython.cast(cython.longlong, mant * 0xffffffffffff))
+            if expo < 0:
+                # A double's exponent is usually limited to [-1024, 1024]
+                expo += 0xFFFF
+            if math.isinf(mant):
```

Comment:
I think you can speed this function access if you declare these as cython globals.